indexing-hadoop/src/main/java/org/apache/druid/indexer/DeterminePartitionsJob.java

```diff
@@ -691,8 +691,9 @@ protected void innerReduce(Context context, SortableBytes keyBytes, Iterable<Dim
             // One more shard to go
             final ShardSpec shardSpec;
 
-            if (currentDimPartition.rows < config.getTargetPartitionSize() * SHARD_COMBINE_THRESHOLD) {
-              // Combine with previous shard
+            if (currentDimPartition.rows < config.getTargetPartitionSize() * SHARD_COMBINE_THRESHOLD &&
+                currentDimPartitions.partitions.size() > 0) {
```

Comment:
Consider `!isEmpty()` over `size() > 0`: https://rules.sonarsource.com/java/tag/clumsy/RSPEC-1155

Reply:
Changed to use !isEmpty()

Reply:
weird, GH didn't pick up my latest commit